src/main/java/io/anserini/kg/freebase/EntityLinking.java

```diff
@@ -0,0 +1,360 @@
+package io.anserini.kg.freebase;
+
+import com.google.common.collect.MinMaxPriorityQueue;
+import io.anserini.kg.freebase.IndexTopics.TopicLuceneDocumentGenerator;
+import io.anserini.rerank.ScoredDocuments;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.lucene.analysis.core.SimpleAnalyzer;
+import org.apache.lucene.index.DirectoryReader;
+import org.apache.lucene.index.IndexReader;
+import org.apache.lucene.queryparser.classic.MultiFieldQueryParser;
+import org.apache.lucene.queryparser.classic.QueryParser;
+import org.apache.lucene.search.IndexSearcher;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.TopDocs;
+import org.apache.lucene.search.similarities.ClassicSimilarity;
+import org.apache.lucene.search.similarities.Similarity;
+import org.apache.lucene.store.FSDirectory;
+import org.kohsuke.args4j.*;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+
+/**
+ * Lookups a Freebase mid and returns all properties associated with it.
+ */
+public class EntityLinking implements Closeable {
+  private static final Logger LOG = LogManager.getLogger(EntityLinking.class);
+
+  private final IndexReader reader;
+
+  private static int numHits = 200;
+
+  static final class Args {
+    // Required arguments
+
+    @Option(name = "-index", metaVar = "[Path]", required = true, usage = "index path")
+    public String index;
+
+    @Option(name = "-data", metaVar = "[file]", required =  true, usage = "dataset file")
+    public String data;
+
+    @Option(name = "-goldData", usage = "supplying with the ground truth dataset")
```

Comment:
This is the only option that does not have a `metaVar`. Also explain if this is a folder or what exactly is expected.